src/plugins/staticRootPlugin/StaticModelProvider.js

```diff
@@ -9,30 +9,89 @@ define([
+                objectLeaf[newIdentifier] = { ...objectLeaf[nodeKey] };
+                delete objectLeaf[nodeKey];
+                objectLeaf[newIdentifier] = parseTreeLeaf(newIdentifier, objectLeaf[newIdentifier], idMap, namespace);
             } else {
-                newId = objectUtils.makeKeyString({
-                    namespace: rootIdentifier.namespace,
-                    key: i
+                objectLeaf[nodeKey] = parseTreeLeaf(nodeKey, objectLeaf[nodeKey], idMap, namespace);
+            }
+        });
+
+        return objectLeaf;
+    }
+
+    function parseArrayLeaf(arrayLeaf, idMap, namespace) {
+        arrayLeaf.forEach((leafValue, index) => {
+            arrayLeaf[index] = parseTreeLeaf(null, leafValue, idMap, namespace);
+        });
+
+        return arrayLeaf;
+    }
+
+    function parseBranchedLeaf(branchedLeafValue, idMap, namespace) {
+        if (Array.isArray(branchedLeafValue)) {
+            return parseArrayLeaf(branchedLeafValue, idMap, namespace);
+        } else {
+            return parseObjectLeaf(branchedLeafValue, idMap, namespace);
+        }
+    }
+
+    function parseTreeLeaf(leafKey, leafValue, idMap, namespace) {
+        const hasChild = typeof leafValue === 'object';
+        if (hasChild) {
+            return parseBranchedLeaf(leafValue, idMap, namespace);
+        }
+
+        if (leafKey === 'key') {
+            return idMap.get(leafValue);
+        } else if (leafKey === 'namespace') {
+            return namespace;
+        } else if (leafKey === 'location') {
+            if (idMap.get(leafValue)) {
+                const newLocationIdentifier = objectUtils.makeKeyString({
+                    namespace,
+                    key: idMap.get(leafValue)
                 });
+
+                return newLocationIdentifier;
             }
 
-            while (objectString.indexOf(originalId) !== -1) {
-                objectString = objectString.replace(
-                    '"' + originalId + '"',
-                    '"' + newId + '"'
-                );
+            return null;
+        } else if (idMap.get(leafValue)) {
+            const newIdentifier = objectUtils.makeKeyString({
+                namespace,
+                key: idMap.get(leafValue)
+            });
+
+            return newIdentifier;
+        } else {
+            return leafValue;
+        }
+    }
+
+    function rewriteObjectIdentifiers(importData, rootIdentifier) {
+        const namespace = rootIdentifier.namespace;
+        const idMap = new Map();
+        const objectTree = importData.openmct;
+
+        Object.keys(objectTree).forEach((originalId, index) => {
+            if (originalId === importData.rootId) {
+                idMap.set(originalId, rootIdentifier.key);
+            } else {
+                idMap.set(originalId, index.toString());
             }
         });
 
-        return JSON.parse(objectString);
+        const newTree = parseTreeLeaf(null, objectTree, idMap, namespace);
```

Comment:
can  just `return parseTreeLeaf(null, objectTree, idMap, namespace);`

Reply:
I like giving it a name here - it makes it more clear what's being returned